### PR TITLE
rbac: update documentation about policy evaluation order

### PR DESCRIPTION
**Pull request by yangminzhu**
Signed-off-by: Yangmin Zhu <ymzhu@google.com>

Commit Message: Update the RBAC API documentation to match the implemented behavior about policy evaluation order.
Additional Description:
Risk Level: N/A, only small doc update
Testing: N/A
Docs Changes: Updated
Release Notes: N/A
Platform Specific Features:
[Optional Runtime guard:]
Optional Fixes https://github.com/envoyproxy/envoy/issues/16152
[Optional Deprecated:]
[Optional [API Considerations](https://github.com/envoyproxy/envoy/blob/main/api/review_checklist.md):]

### CODE REVIEW

File: api/envoy/config/rbac/v3/rbac.proto [hunk 1]
```diff
@@ -105,6 +105,8 @@ message RBAC {
   Action action = 1 [(validate.rules).enum = {defined_only: true}];
 
   // Maps from policy name to policy. A match occurs when at least one policy matches the request.
+  // The policies are evaluated in the order of the policy name. The policy with the smallest name
```
Comment (zuercher) [inline api/envoy/config/rbac/v3/rbac.proto:108]: Just a drive-by, but I think we should be clear about what "smallest" means. Should it say lexicographic order? e.g. "The policies are evaluated in lexicographic order."

Comment (yangminzhu) [reply api/envoy/config/rbac/v3/rbac.proto]: thanks for the comment, updated.

**Review by zuercher — APPROVED** [review]
Thanks!
